src/shogun2-core/src/main/java/de/terrestris/shogun2/dao/GenericHibernateDao.java

```diff
@@ -96,6 +96,37 @@ public E findById(ID id) {
 	}
 
 	/**
+	 * Returns a list of entity objects that have field named
+	 * <code>fieldName</code>, which has an object <code>fieldEntity</code>
+	 * as value.
+	 *
+	 * @param fieldName The name of the field
+	 * @param fieldEntity The element that should be set as value
+	 * @param criterion Additional criterions to apply (optional)
+	 *
+	 * @return The list of objects
+	 */
+	@SuppressWarnings("unchecked")
+	public List<E> findAllWhereFieldEquals(String fieldName, Object fieldEntity,
```

Comment:
What happens if `fieldEntity` is `null`?

I would assume that `isField` becomes `false` and one runs into the `IllegalArgumentException` (at least if you take my suggestion of also checking the type).

But this is a valid question, right? Please find me all instances of `E` where the field named `fieldName` is (or is not) `null`. What do you think?

I think we could branch off to a dfifferent method in the service, but this comes from the top of my head.